### PR TITLE
Rename ORANGE construction files

### PR DESCRIPTION
**Pull request by sethrj**
This rearranges the ORANGE construction directory, moving the components used for "shape construction" (e.g. the CSG tree) into `orangeinp/`, and moving the `OrangeInput` class to the main directory. The CSG renaming solves all the weird namespace issues of `using namespace orange::csg`.

### CODE REVIEW

File: app/orange-update.cc [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #    include <fstream>
 #    include <nlohmann/json.hpp>
 
-#    include "orange/construct/OrangeInputIO.json.hh"
+#    include "orange/OrangeInputIO.json.hh"
```
Comment (elliottbiondo) [inline app/orange-update.cc:25]: This .cc file doesn't have any info about what version of ORANGE geometries are being updated. Though unrelated to this MR, consider adding a short description.

File: src/orange/detail/BIHBuilder.hh [hunk 2]
```diff
@@ -4,6 +4,7 @@
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
 //! \file orange/detail/BIHBuilder.hh
+//! \todo move to orangeinp/BihBuilder
```
Comment (elliottbiondo) [inline src/orange/detail/BIHBuilder.hh]: Remove

File: src/orange/detail/BIHPartitioner.hh [hunk 3]
```diff
@@ -4,6 +4,7 @@
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
 //! \file orange/detail/BIHPartitioner.hh
+//! \todo Move to orangeinp/detail/BihPartitioner
```
Comment (elliottbiondo) [inline src/orange/detail/BIHPartitioner.hh]: Remove

**Review by elliottbiondo — APPROVED** [review]

File: app/orange-update.cc (hunk 1 above)
Comment (sethrj) [reply app/orange-update.cc:25]: Yeah, this just reads in a JSON file (which will translate any "old" parameter names, etc.) and writes it back out again. I'll add a brief in this file, but the app *is* actually documented in `doc/main/usage.rst`.

**Pull request merged**